tests/bsim/bluetooth/host/misc/sample_test/src/dut.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+
+#include "testlib/conn.h"
+#include "testlib/scan.h"
+
+#include "bs_helpers/flags.h"
+#include "bs_helpers/sync.h"
+#include "bs_helpers/testcase.h"
+#include "bs_helpers/timeout.h"
+
+/* local includes */
+#include "data.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(dut, CONFIG_TEST_APP_LOG_LEVEL);
+
+static DEFINE_FLAG(is_subscribed);
+
+static void ccc_changed(const struct bt_gatt_attr *attr, uint16_t value)
+{
+	/* assume we only get it for the `test_gatt_service` */
+	if (value) {
```

Comment:
```suggestion
	/* assume we only get it for the `test_gatt_service` */
	if (value != 0) {
```

Our coding guidelines refer to MISRA which has a rule that boolean expressions (like `if`) should use boolean values